packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js

```diff
@@ -44,13 +46,22 @@ if ('Success' == queryResponseCode.toString() || 'PartialSuccess' == queryRespon
 			// Files are stored in format: <CX_ID>/<PATIENT_ID>/<CX_ID>_<PATIENT_ID>_<DOC_ID>
 			var fileName = [request.cxId, request.patientResourceId, attachment.documentUniqueId + '.b64'].join('_');
 			var filePath = [request.cxId, request.patientResourceId, fileName].join('/');
+      var docExists = xcaReadFromFile(filePath.toString());
 
 			try {
 
 				attachment.contentType = entry.*::mimeType.toString();
+				attachment.fileName = filePath.toString();
 				attachment.url = filePath.toString();
+
+        if (docExists) {
+          attachment.isNew = false;
+        } else {
+          attachment.isNew = true;
+        }
+
```

Comment:
The updates in this file need to be here and not on the api because it is here that we can actually see if the file exists before inserting in the s3. If we leave this to the api the doc will have already been uploaded and will always say it exists